src/middleware/maintenance.middleware.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable, NestMiddleware, HttpStatus, } from '@nestjs/common';
+import { firstValueFrom } from "rxjs";
+import { Request, Response, NextFunction } from 'express';
+import { ConfigService } from '@nestjs/config';
+import { HttpService } from "@nestjs/axios";
+import { EaseyException } from "../exceptions";
+
+@Injectable()
+export class MaintenanceMiddleware implements NestMiddleware {
+
+  private readonly errorMessage: string = "The server is temporarily unable to service your request due to maintenance. Please try again later."
+
+  constructor(private readonly configService: ConfigService, private readonly httpService: HttpService) { }
+
+  async use(request: Request, res: Response, next: NextFunction) {
+    try {
+      const url = this.configService.get("app.authApi").uri + "/authentication/login-state";
+      const apiKey = this.configService.get("app.apiKey") || request.headers["x-api-key"];
+
+      const forwardedForHeader = request.headers["x-forwarded-for"];
+      let ip = request.ip;
+      if (forwardedForHeader !== null && forwardedForHeader !== undefined) {
+        ip = (forwardedForHeader as string)?.split(",")[0];
+      }
+
+      if (!apiKey) {
+        throw new EaseyException(new Error('API Key required via "x-api-key" request header!'), HttpStatus.BAD_REQUEST);
+      }
```

Comment:
Do we actually want to check this here? I don't think the API key is required when running with `*_ENABLE_AUTH_TOKEN` is false. I know this will only affect running the project locally, but may be better to punt the responsibility of validating the key.